### PR TITLE
doc: add 12.0.1 release notes

### PR DESCRIPTION
**Pull request by theanalyst**
Adding the 12.0.1 release notes, also added all the pending changes from 12.0.0

### CODE REVIEW

File: doc/release-notes.rst [hunk 1]
```diff
@@ -1,11 +1,544 @@
 ===============
  Release Notes
 ===============
+
+v12.0.1 Luminous (dev)
+====================
```
Comment (tchaikov) [inline doc/release-notes.rst]: nit, might want to add two more `==` here, so sphinx-build won't complain.

File: doc/release-notes.rst [hunk 2]
```diff
@@ -1,11 +1,544 @@
...
+
+This is the second development checkpoint release of Luminous, the next long term
+stable release.
+
+Major changes from 12.0.0
+------------------------
```
Comment (tchaikov) [inline doc/release-notes.rst]: nit, might want to add one more `-` here, so sphinx-build won't complain.

File: doc/release-notes.rst (hunk 1 above)
Comment (theanalyst) [inline doc/release-notes.rst]: ack

**Comment by theanalyst** [comment]
test this please

**Pull request merged**